// graphql-scalars/dist/commonjs/resolvers/BigInt.js
"use strict";
// Taken from https://github.com/stems/graphql-bigint/
Object.defineProperty(exports, "__esModule", { value: true });
const definition_1 = require("graphql/type/definition");
const kinds_1 = require("graphql/language/kinds");
const MAX_INT = Number.MAX_SAFE_INTEGER;
const MIN_INT = Number.MIN_SAFE_INTEGER;
exports.default = new definition_1.GraphQLScalarType({
    name: 'BigInt',
    description: 'The `BigInt` scalar type represents non-fractional signed whole numeric ' +
        'values. BigInt can represent values between -(2^53) + 1 and 2^53 - 1. ',
    serialize: coerceBigInt,
    parseValue: coerceBigInt,
    parseLiteral(ast) {
        if (ast.kind === kinds_1.Kind.INT) {
            const num = parseInt(ast.value, 10);
            if (num <= MAX_INT && num >= MIN_INT) {
                return num;
            }
        }
        return null;
    }
});
function coerceBigInt(value) {
    if (value === '') {
        throw new TypeError('BigInt cannot represent non 53-bit signed integer value: (empty string)');
    }
    const num = Number(value);
    if (num !== num || num > MAX_INT || num < MIN_INT) {
        throw new TypeError('BigInt cannot represent non 53-bit signed integer value: ' + String(value));
    }
    const int = Math.floor(num);
    if (int !== num) {
        throw new TypeError('BigInt cannot represent non-integer value: ' + String(value));
    }
    return int;
}
