// graphql-scalars/dist/commonjs/resolvers/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
/// <reference path="../../types/graphql-type-json.d.ts" />
const DateTime_1 = require("./DateTime");
exports.DateTime = DateTime_1.default;
const NonPositiveInt_1 = require("./NonPositiveInt");
exports.NonPositiveInt = NonPositiveInt_1.default;
const PositiveInt_1 = require("./PositiveInt");
exports.PositiveInt = PositiveInt_1.default;
const NonNegativeInt_1 = require("./NonNegativeInt");
exports.NonNegativeInt = NonNegativeInt_1.default;
exports.UnsignedInt = NonNegativeInt_1.default;
const NegativeInt_1 = require("./NegativeInt");
exports.NegativeInt = NegativeInt_1.default;
const NonPositiveFloat_1 = require("./NonPositiveFloat");
exports.NonPositiveFloat = NonPositiveFloat_1.default;
const PositiveFloat_1 = require("./PositiveFloat");
exports.PositiveFloat = PositiveFloat_1.default;
const NonNegativeFloat_1 = require("./NonNegativeFloat");
exports.NonNegativeFloat = NonNegativeFloat_1.default;
exports.UnsignedFloat = NonNegativeFloat_1.default;
const NegativeFloat_1 = require("./NegativeFloat");
exports.NegativeFloat = NegativeFloat_1.default;
const EmailAddress_1 = require("./EmailAddress");
exports.EmailAddress = EmailAddress_1.default;
const URL_1 = require("./URL");
exports.URL = URL_1.default;
const PhoneNumber_1 = require("./PhoneNumber");
exports.PhoneNumber = PhoneNumber_1.default;
const PostalCode_1 = require("./PostalCode");
exports.PostalCode = PostalCode_1.default;
const BigInt_1 = require("./BigInt");
exports.BigInt = BigInt_1.default;
exports.Long = BigInt_1.default;
const GUID_1 = require("./GUID");
exports.GUID = GUID_1.default;
const HexColorCode_1 = require("./HexColorCode");
exports.HexColorCode = HexColorCode_1.default;
const HSL_1 = require("./HSL");
exports.HSL = HSL_1.default;
const HSLA_1 = require("./HSLA");
exports.HSLA = HSLA_1.default;
const IPv4_1 = require("./IPv4");
exports.IPv4 = IPv4_1.default;
const IPv6_1 = require("./IPv6");
exports.IPv6 = IPv6_1.default;
const ISBN_1 = require("./ISBN");
exports.ISBN = ISBN_1.default;
const MAC_1 = require("./MAC");
exports.MAC = MAC_1.default;
const Port_1 = require("./Port");
exports.Port = Port_1.default;
const RGB_1 = require("./RGB");
exports.RGB = RGB_1.default;
const RGBA_1 = require("./RGBA");
exports.RGBA = RGBA_1.default;
const USCurrency_1 = require("./USCurrency");
exports.USCurrency = USCurrency_1.default;
const graphql_type_json_1 = require("graphql-type-json");
exports.JSON = graphql_type_json_1.GraphQLJSON;
exports.JSONObject = graphql_type_json_1.GraphQLJSONObject;
