// graphql-scalars/dist/commonjs/resolvers/URL.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const graphql_1 = require("graphql");
const error_1 = require("graphql/error");
const language_1 = require("graphql/language");
if (typeof URL === 'undefined') {
    global['URL'] = require('url').URL;
}
exports.default = new graphql_1.GraphQLScalarType({
    name: 'URL',
    description: 'A field whose value conforms to the standard URL format as specified in RFC3986: https://www.ietf.org/rfc/rfc3986.txt.',
    serialize(value) {
        if (typeof value !== 'string' && !(value instanceof URL)) {
            throw new TypeError(`Value is not string or URL: ${value}`);
        }
        return new URL(value.toString()).toString();
    },
    parseValue(value) {
        if (typeof value !== 'string') {
            throw new TypeError(`Value is not string: ${value}`);
        }
        return new URL(value);
    },
    parseLiteral(ast) {
        if (ast.kind !== language_1.Kind.STRING) {
            throw new error_1.GraphQLError(`Can only validate strings as URLs but got a: ${ast.kind}`);
        }
        return new URL(ast.value);
    },
});
