// graphql-scalars/dist/commonjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var typeDefs_1 = require("./typeDefs");
exports.DateTimeTypeDefinition = typeDefs_1.DateTime;
exports.EmailAddressTypeDefinition = typeDefs_1.EmailAddress;
exports.NegativeFloatTypeDefinition = typeDefs_1.NegativeFloat;
exports.NegativeIntTypeDefinition = typeDefs_1.NegativeInt;
exports.NonNegativeFloatTypeDefinition = typeDefs_1.NonNegativeFloat;
exports.NonNegativeIntTypeDefinition = typeDefs_1.NonNegativeInt;
exports.NonPositiveFloatTypeDefinition = typeDefs_1.NonPositiveFloat;
exports.NonPositiveIntTypeDefinition = typeDefs_1.NonPositiveInt;
exports.PhoneNumberTypeDefinition = typeDefs_1.PhoneNumber;
exports.PositiveFloatTypeDefinition = typeDefs_1.PositiveFloat;
exports.PositiveIntTypeDefinition = typeDefs_1.PositiveInt;
exports.PostalCodeTypeDefinition = typeDefs_1.PostalCode;
exports.UnsignedFloatTypeDefinition = typeDefs_1.UnsignedFloat;
exports.UnsignedIntTypeDefinition = typeDefs_1.UnsignedInt;
exports.URLTypeDefinition = typeDefs_1.URL;
exports.BigIntTypeDefinition = typeDefs_1.BigInt;
exports.LongTypeDefinition = typeDefs_1.Long;
exports.GUIDDefinition = typeDefs_1.GUID;
exports.HexColorCodeDefinition = typeDefs_1.HexColorCode;
exports.HSLDefinition = typeDefs_1.HSL;
exports.HSLADefinition = typeDefs_1.HSLA;
exports.IPv4Definition = typeDefs_1.IPv4;
exports.IPv6Definition = typeDefs_1.IPv6;
exports.ISBNDefinition = typeDefs_1.ISBN;
exports.MACDefinition = typeDefs_1.MAC;
exports.PortDefinition = typeDefs_1.Port;
exports.RGBDefinition = typeDefs_1.RGB;
exports.RGBADefinition = typeDefs_1.RGBA;
exports.USCurrencyDefinition = typeDefs_1.USCurrency;
exports.JSONDefinition = typeDefs_1.JSON;
exports.JSONObjectDefinition = typeDefs_1.JSONObject;
var typeDefs_2 = require("./typeDefs");
exports.typeDefs = typeDefs_2.default;
var resolvers_1 = require("./resolvers");
exports.DateTimeResolver = resolvers_1.DateTime;
exports.EmailAddressResolver = resolvers_1.EmailAddress;
exports.NegativeFloatResolver = resolvers_1.NegativeFloat;
exports.NegativeIntResolver = resolvers_1.NegativeInt;
exports.NonNegativeFloatResolver = resolvers_1.NonNegativeFloat;
exports.NonNegativeIntResolver = resolvers_1.NonNegativeInt;
exports.NonPositiveFloatResolver = resolvers_1.NonPositiveFloat;
exports.NonPositiveIntResolver = resolvers_1.NonPositiveInt;
exports.PhoneNumberResolver = resolvers_1.PhoneNumber;
exports.PositiveFloatResolver = resolvers_1.PositiveFloat;
exports.PositiveIntResolver = resolvers_1.PositiveInt;
exports.PostalCodeResolver = resolvers_1.PostalCode;
exports.UnsignedFloatResolver = resolvers_1.UnsignedFloat;
exports.UnsignedIntResolver = resolvers_1.UnsignedInt;
exports.URLResolver = resolvers_1.URL;
exports.BigIntResolver = resolvers_1.BigInt;
exports.LongResolver = resolvers_1.Long;
exports.GUIDResolver = resolvers_1.GUID;
exports.HexColorCodeResolver = resolvers_1.HexColorCode;
exports.HSLResolver = resolvers_1.HSL;
exports.HSLAResolver = resolvers_1.HSLA;
exports.IPv4Resolver = resolvers_1.IPv4;
exports.IPv6Resolver = resolvers_1.IPv6;
exports.ISBNResolver = resolvers_1.ISBN;
exports.MACResolver = resolvers_1.MAC;
exports.PortResolver = resolvers_1.Port;
exports.RGBResolver = resolvers_1.RGB;
exports.RGBAResolver = resolvers_1.RGBA;
exports.USCurrencyResolver = resolvers_1.USCurrency;
exports.JSONResolver = resolvers_1.JSON;
exports.JSONObjectResolver = resolvers_1.JSONObject;
const resolvers = require("./resolvers");
exports.resolvers = resolvers;
var mocks_1 = require("./mocks");
exports.DateTimeMock = mocks_1.DateTime;
exports.EmailAddressMock = mocks_1.EmailAddress;
exports.NegativeFloatMock = mocks_1.NegativeFloat;
exports.NegativeIntMock = mocks_1.NegativeInt;
exports.NonNegativeFloatMock = mocks_1.NonNegativeFloat;
exports.NonNegativeIntMock = mocks_1.NonNegativeInt;
exports.NonPositiveFloatMock = mocks_1.NonPositiveFloat;
exports.NonPositiveIntMock = mocks_1.NonPositiveInt;
exports.PhoneNumberMock = mocks_1.PhoneNumber;
exports.PositiveFloatMock = mocks_1.PositiveFloat;
exports.PositiveIntMock = mocks_1.PositiveInt;
exports.PostalCodeMock = mocks_1.PostalCode;
exports.UnsignedFloatMock = mocks_1.UnsignedFloat;
exports.UnsignedIntMock = mocks_1.UnsignedInt;
exports.URLMock = mocks_1.URL;
exports.BigIntMock = mocks_1.BigInt;
exports.LongMock = mocks_1.Long;
exports.GUIDMock = mocks_1.GUID;
exports.HexColorCodeMock = mocks_1.HexColorCode;
exports.HSLMock = mocks_1.HSL;
exports.HSLAMock = mocks_1.HSLA;
exports.IPv4Mock = mocks_1.IPv4;
exports.IPv6Mock = mocks_1.IPv6;
exports.ISBNMock = mocks_1.ISBN;
exports.MACMock = mocks_1.MAC;
exports.PortMock = mocks_1.Port;
exports.RGBMock = mocks_1.RGB;
exports.RGBAMock = mocks_1.RGBA;
exports.USCurrencyMock = mocks_1.USCurrency;
exports.JSONMock = mocks_1.JSON;
exports.JSONObjectMock = mocks_1.JSONObject;
const mocks = require("./mocks");
exports.mocks = mocks;
var RegularExpression_1 = require("./RegularExpression");
exports.RegularExpression = RegularExpression_1.default;
