// graphql-scalars/dist/commonjs/typeDefs.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DateTime = 'scalar DateTime';
exports.EmailAddress = 'scalar EmailAddress';
exports.NegativeFloat = 'scalar NegativeFloat';
exports.NegativeInt = 'scalar NegativeInt';
exports.NonNegativeFloat = 'scalar NonNegativeFloat';
exports.NonNegativeInt = 'scalar NonNegativeInt';
exports.NonPositiveFloat = 'scalar NonPositiveFloat';
exports.NonPositiveInt = 'scalar NonPositiveInt';
exports.PhoneNumber = 'scalar PhoneNumber';
exports.PositiveFloat = 'scalar PositiveFloat';
exports.PositiveInt = 'scalar PositiveInt';
exports.PostalCode = 'scalar PostalCode';
exports.UnsignedFloat = 'scalar UnsignedFloat';
exports.UnsignedInt = 'scalar UnsignedInt';
exports.URL = 'scalar URL';
exports.BigInt = 'scalar BigInt';
exports.Long = 'scalar Long';
exports.GUID = `scalar GUID`;
exports.HexColorCode = `scalar HexColorCode`;
exports.HSL = `scalar HSL`;
exports.HSLA = `scalar HSLA`;
exports.IPv4 = `scalar IPv4`;
exports.IPv6 = `scalar IPv6`;
exports.ISBN = `scalar ISBN`;
exports.MAC = `scalar MAC`;
exports.Port = `scalar Port`;
exports.RGB = `scalar RGB`;
exports.RGBA = `scalar RGBA`;
exports.USCurrency = `scalar USCurrency`;
exports.JSON = `scalar JSON`;
exports.JSONObject = `scalar JSONObject`;
exports.default = [
    exports.DateTime,
    exports.EmailAddress,
    exports.NegativeFloat,
    exports.NegativeInt,
    exports.NonNegativeFloat,
    exports.NonNegativeInt,
    exports.NonPositiveFloat,
    exports.NonPositiveInt,
    exports.PhoneNumber,
    exports.PositiveFloat,
    exports.PositiveInt,
    exports.PostalCode,
    exports.UnsignedFloat,
    exports.UnsignedInt,
    exports.URL,
    exports.BigInt,
    exports.Long,
    exports.GUID,
    exports.HexColorCode,
    exports.HSL,
    exports.HSLA,
    exports.IPv4,
    exports.IPv6,
    exports.ISBN,
    exports.MAC,
    exports.Port,
    exports.RGB,
    exports.RGBA,
    exports.USCurrency,
    exports.JSON,
    exports.JSONObject,
];
